services/director-v2/src/simcore_service_director_v2/models/domains/comp_tasks.py

```diff
@@ -22,6 +23,10 @@ class ComputationTaskCreate(BaseModel):
     start_pipeline: Optional[bool] = Field(
         False, description="if True the computation pipeline will start right away"
     )
+    subgraph: Optional[List[NodeID]] = Field(
+        None,
+        description="An optional set of nodes that must be executed, if empty the whole pipeline is executed",
+    )
```

Comment:
From what I understand the frontend would select a sub portion of nodes to be executed which are stored in this filed? These are also used in the filtering.

**The below suggestion is based on the assumption that the answer to the previous question is YES**
---

Sorry for the following late suggestion, I really liked your idea and the below should improve with added benefits:
- possibility to fully run the pipeline without frontend interaction
- a more controlled run procedure (backend centric)
- easier API integration for users (while running the pipeline or parts of it)


I would say that it would be more convenient to be able to store multiple subgraphs (the ones which make sense for the pipeline). I would also encode order of execution and priority of node groups. Example below.

Say we have the following 3 groups of nodes:

```python
subgraphs: Dict[int, List[List[str]]] = {
    0: [
        ["comp_pipeline_node_1", "comp_pipeline_node_2"],
        ["comp_pipeline_node_3", "comp_pipeline_node_4"],
    ],
    1: [["comp_pipeline_node_5"]],
    2: [["comp_pipeline_node_6"]],
}
```

Where the (integer) index is the priority and the (lists of nodes) in the list represent the groups of subgraphs meant to be ran in parallel.

In the above example when the **Run button** is pressed:
- the subgraphs (**2 in this case**) at index `0` will be executed in parallel
- when they are finished, the subgraphs (**1 in this case**) at index `1` will be executed in parallel
- when they are finished, the subgraphs (**1 in this case**) at index `2` will be executed in parallel 
- process is completed as all indexes from the dictionary have finished.

To fully support the guided mode at this point you will need an extra (integer) field which points out which index to execute. If the value is `None` all subgraphs will be ran in the above order.

---

I would at least have this structure in place in the backend even if it will not be fully implemented as I think will prove to be more flexible in the future.

Reply:
After a discussion with @GitHK upgraded the description.